### PR TITLE
Zephyr Skeleton for TL321x

### PR DESCRIPTION
**Pull request by damien0x0023**
soc: riscv: telink: telink_tlx: Create SoC Zephyr Skeleton for TL321x

1. Add new SoC Telink TLx(telink_tlx) based on Telink B9x(telink_b9x).
2. Add new board tl3218x based on tlsr9528a.
3. Add Zephyr driver and DTS files for TLx SoC.
4. Add tl3218x.& tl3218x.overlay in ot-cli samples to apply uart.
5. Update hal_telink to match tl321x driver skeleton.
6. Update openthread_telink_lib to match new SoC series.
    
Known issues:
    
1. USB driver does not work normally.
    usb-cdm feature needs to be applied.
3. Remove Deep Retention samples.
    They may not be able to build due to missing definitions.

### CODE REVIEW

**Comment by andriy-bilynskyy** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/36283106d9b0b1d5fb3025a98be5a58007ac3d77 (develop,telink_tl321x) | https://github.com/telink-semi/hal_telink/commit/65648c632e2adeab955c6f6743ad47ad646c5c6e (tl321x_skeleton) | [telink-semi/hal_telink@36283106..65648c63](https://github.com/telink-semi/hal_telink/compare/36283106d9b0b1d5fb3025a98be5a58007ac3d77..65648c632e2adeab955c6f6743ad47ad646c5c6e) |
| openthread_telink_lib | https://github.com/telink-semi/openthread_telink_lib/commit/80645ba0b4b589afd8c0712cb1a01091f4b35ab8 (main) | https://github.com/telink-semi/openthread_telink_lib/commit/84e12fd24613b2871c47378c39c2f44447f528ba (tl321x_skeleton) | [telink-semi/openthread_telink_lib@80645ba0..84e12fd2](https://github.com/telink-semi/openthread_telink_lib/compare/80645ba0b4b589afd8c0712cb1a01091f4b35ab8..84e12fd24613b2871c47378c39c2f44447f528ba) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

**Comment by liucoldstar** [comment]
@s07641069 

Could you take a quick look and see if there is any improvement needed?

Andy

File: boards/riscv/tl321x/Kconfig.board
Comment (s07641069) [inline boards/riscv/tl321x/Kconfig.board]: In all new files use 2024
In modified file use XXXX-2024

File: boards/riscv/tl321x/Kconfig.defconfig
Comment (s07641069) [inline boards/riscv/tl321x/Kconfig.defconfig]: Find globally in all new files keys like
b9
tlsr9
And replace or remove them

File: boards/riscv/tl321x/doc/index.rst
Comment (s07641069) [inline boards/riscv/tl321x/doc/index.rst]: same here and in some more files old copy pasted names of board/SoC

File: boards/riscv/tl321x/tl3218x-pinctrl.dtsi
Comment (s07641069) [inline boards/riscv/tl321x/tl3218x-pinctrl.dtsi]: B9x , B92?

File: drivers/bluetooth/hci/hci_tlx.c
Comment (s07641069) [inline drivers/bluetooth/hci/hci_tlx.c]: looks like you want to add new SoC, but keep old build like fake B92

**Review by s07641069 — CHANGES_REQUESTED** [review]

File: drivers/bluetooth/hci/hci_tlx.c
Comment (s07641069) [reply drivers/bluetooth/hci/hci_tlx.c]: @andriy-bilynskyy @liucoldstar @damien0x0023 
In such case try to update/adopt new files as much as possible to simplify next porting
And we will take a look carefully only into changes file which can impact others SoCs
To avoid lots of PRs for each module/driver, maybe will be easily keep this PR and modules PRs till you will have more changes and after will merge all together into develop branches ?

File: boards/riscv/tl321x/doc/index.rst
Comment (liucoldstar) [reply boards/riscv/tl321x/doc/index.rst]: Changes to these contents have been planned. Will update them in another PR.

File: boards/riscv/tl321x/tl3218x-pinctrl.dtsi
Comment (liucoldstar) [reply boards/riscv/tl321x/tl3218x-pinctrl.dtsi]: Damien will make these changes in another PR.

File: boards/riscv/tl321x/Kconfig.board
Comment (damien0x0023) [reply boards/riscv/tl321x/Kconfig.board]: Okay, will update year with related in the following commits.

File: boards/riscv/tl321x/Kconfig.defconfig
Comment (damien0x0023) [reply boards/riscv/tl321x/Kconfig.defconfig]: Understood. Will replace them one by one later.

File: drivers/bluetooth/hci/hci_tlx.c
Comment (damien0x0023) [reply drivers/bluetooth/hci/hci_tlx.c]: Thank you very much for your suggestions!
We will do more modificaiton and improvement based on your comments in andy's working branch tl321x_bringup.

This PR and my working branch tl321x_skeleton may be expired so we can ignore it.
Please DO NOT MERGE this PR because it's only very initial draft.

**Comment by damien0x0023** [comment]
Adopt pinctrl and uart driver
    
    - update pinctrl driver.
    - update uart driver to enable serial output.

**Pull request closed without merging**
